src/archivist/recorder/repositories/git/git.js

```diff
@@ -114,6 +104,18 @@ export default class Git {
     return this.git.raw(options);
   }
 
+  async show(options) {
+    return this.git.show(options);
+  }
+
+  async getFullHash(shortHash) {
+    return (await this.git.show([ shortHash, '--pretty=%H', '-s' ])).trim();
```

Comment:
I don't understand how that would work in cases where the short hash is ambiguous 🤔

Reply:
It won't… and it can't. That's why I wanted to switch to full hash.

Reply:
I get that, but the way we get the full hash is by getting the source commit through its short hash, so won't a collision always prevent `getFullHash` from working?

Reply:
We agreed on a synchronous call to have a better implementation in a separate PR.